perun/postprocess/kernel_regression/run.py

```diff
@@ -0,0 +1,555 @@
+
+        **Bandwidth Selection** is the most important factor at each approach of kernel regression, since
+        this value significantly affects the smoothness of the resulting estimate. In case, when is choose the
+        inappropriate value, in the most cases can be expected the following two situations. The **small**
+        bandwidth value reproduce estimated data and vice versa, the **large** value leads to over-leaving, so
+        to average of the estimated data. Therefore are used the methods to determine the bandwidth value.
+        One of the most widespread and most commonly used methods is the **cross-validation** method. This
+        method is based on the estimate of the regression function in which will be omitted *i-th* observation.
+        In this postprocessor is this method available in the **estimator-setting** mode. Another methods to
+        determine the bandwidth, which are available in the remaining modes of this postprocessor are **scott**
+        and **silverman** method. More information about these methods and its definition you cas see in the
+        part :ref:`postprocessors-kernel-regression-method_selection`.
+
+    This postprocessor in summary offers five different modes, which does not differ in the resulting estimate,
+    but in the way of computation the resulting estimate. Better said, it means, that the result of each mode
+    is the **kernel estimate** with relevant parameters, selected according to the concrete mode. In short we
+    will describe the individual methods, for more information about it, you can visit the relevant parts of
+    documentation:
+
+        * **Estimator-Settings**: Nadaraya-Watson kernel regression with specific settings for estimate
+        * **User-Selection**: Nadaraya-Watson kernel regression with user bandwidth
+        * **Method-Selection**: Nadaraya-Watson kernel regression with supporting bandwidth selection method
+        * **Kernel-Smoothing**: Kernel regression with different types of kernel and regression methods
+        * **Kernel-Ridge**: Nadaraya-Watson kernel regression with automatic bandwidth selection
+
+    For more details about this approach of non-parametric analysis refer to :ref:`postprocessors-kernel-regression`.
+    """
+    # running default mode with use EstimatorSettings and its default parameters
+    if ctx.invoked_subcommand is None:
+        ctx.invoke(estimator_settings)
+
+
+# Supported modes at executing kernel regression:
+# - estimator-settings: with use EstimatorSettings and its arguments
+# - user-selection: bandwidth defined by user itself
+# - bandwidth_methods: bandwidth computed by helper method for its determine
+# - kernel-smoothing: provides the ability to choose a kernel and other methods
+# - kernel-ridge: set bandwidth by minimizing mean-squared error in leave-one-out cross validation from given range
+_SUPPORTED_MODES = [estimator_settings, user_selection, method_selection, kernel_smoothing, kernel_ridge]
```

Comment:
Suggestion: I will leave the comment here. Since we have 5 subcommands, I would brainstorm about some different names (in particular for first three).